test/v2/core/minter.spec.js

```diff
@@ -0,0 +1,212 @@
+    expect(afterBalance).to.eq(prevBalance.add(parseEther("1")));
+  });
+
+  it("depositFor", async () => {
+    const prevBalance = await sgEth.balanceOf(alice.address);
+    await minter.depositFor(alice.address, {
+      value: parseEther("1"),
+    });
+    const afterBalance = await sgEth.balanceOf(alice.address);
+    expect(afterBalance).to.eq(prevBalance.add(parseEther("1")));
+  });
+
+  it("depositAndStake", async () => {
+    const prevStake = await wsgEth.maxRedeem(deployer.address);
+    const prevBalance = await sgEth.balanceOf(wsgEth.address);
+    await minter.depositAndStake({
+      value: parseEther("1"),
+    });
+    const afterBalance = await sgEth.balanceOf(wsgEth.address);
+    expect(afterBalance).to.eq(prevBalance.add(parseEther("1")));
+
+    const afterStake = await wsgEth.maxRedeem(deployer.address);
+    expect(afterStake).to.eq(prevStake.add(parseEther("1")));
+  });
+
+  it("depositAndStakeFor", async () => {
+    const prevStake = await wsgEth.maxRedeem(alice.address);
+    const prevBalance = await sgEth.balanceOf(wsgEth.address);
+    await minter.depositAndStakeFor(alice.address, {
+      value: parseEther("1"),
+    });
+    const afterBalance = await sgEth.balanceOf(wsgEth.address);
+    expect(afterBalance).to.eq(prevBalance.add(parseEther("1")));
+
+    const afterStake = await wsgEth.maxRedeem(alice.address);
+    expect(afterStake).to.eq(prevStake.add(parseEther("1")));
+  });
+
+  it("withdraw, withdrawTo", async () => {
+    await minter.connect(alice).deposit({
+      value: parseEther("1"),
+    });
+    await expect(minter.connect(alice).withdraw(parseEther("1.1"))).to.be.revertedWith("");
+
+    let prevBalance = await sgEth.balanceOf(alice.address);
+    await minter.connect(alice).withdraw(parseEther("0.5"));
+    let afterBalance = await sgEth.balanceOf(alice.address);
+
+    expect(afterBalance).to.eq(prevBalance.sub(parseEther("0.5")));
+
+    prevBalance = await sgEth.balanceOf(alice.address);
+    await minter.connect(alice).withdrawTo(parseEther("0.5"), alice.address);
+    afterBalance = await sgEth.balanceOf(alice.address);
+
+    expect(afterBalance).to.eq(prevBalance.sub(parseEther("0.5")));
+  });
+
+  it("unstakeAndWithdraw", async () => {
+    await minter.connect(alice).depositAndStake({
+      value: parseEther("1"),
+    });
+    await expect(minter.connect(alice).unstakeAndWithdraw(parseEther("1.1"), alice.address)).to.be.revertedWith("");
+    await expect(minter.connect(alice).unstakeAndWithdraw(parseEther("0.5"), alice.address)).to.be.revertedWith("");
+
+    await wsgEth.connect(alice).approve(minter.address, ethers.constants.MaxUint256);
+    let prevBalance = await wsgEth.balanceOf(alice.address);
+    await minter.connect(alice).unstakeAndWithdraw(parseEther("0.5"), alice.address);
+    let afterBalance = await wsgEth.balanceOf(alice.address);
+
+    expect(afterBalance).to.eq(prevBalance.sub(parseEther("0.5")));
+  });
+
```

Comment:
can we add a comment here denoting all the functions below are Admin or Keeper functions? 
And that we are checking that the auth is safe on them?

Reply:
What's the admin or keeper functions? and I'm not sure what are you saying with auth. you mean ROLES  such as ADMIN, GOV and NOR?